### DISCUSSION TITLE
Provisionally Approved accessibility and component questions

### DISCUSSION THREAD

**Discussion by lgalloherran** [twilio-labs/paste#1446] [Office hours] [answered]
Hey Dsys, 
I'm working on a project related to a new state for Regulatory Compliances bundles and I have some questions. 
[**For the Bundle's table page**](https://www.figma.com/proto/19GxmZa01LDz8LX4u3afQR/Provisionally-Approved?page-id=1090%3A20351&node-id=1090%3A20965&viewport=629%2C361%2C0.05655873194336891&scaling=min-zoom): 
- Status Details: Is this format appropriate for screen readers? 
- Table header: Is there an option to enable the header sections to organize the table content (Like the filter in Excel?)
- Filter: Does it affects the accessibility part the fact of not having a 'filter' button? Is there a pattern for this?

[**For the Instance page**](https://www.figma.com/proto/19GxmZa01LDz8LX4u3afQR/Provisionally-Approved?page-id=1155%3A39496&node-id=1155%3A40131&viewport=236%2C530%2C0.1910802721977234&scaling=min-zoom):
- Call to action on the form of a menu displaying three actions: We are migrating this to the Paste components, but I'm not sure if this action menu exists in Paste, or if there is an alternative to have 3 action items in one row of a table.

**Comment by serifluous** [comment, marked as answer]
Thanks for sharing, @lgalloherran ! We can chat more about your questions tomorrow, but here are some initial answers:

> Status Details: Is this format appropriate for screen readers?

(longer comment in Figma but essentially:) The experience might actually be fine if you gave the icons text that the screen reader would be able to read out: https://paste.twilio.design/icons/usage-guidelines#give-informative-icons-a-title That text would draw a clearer distinction between what the warning icon and success icon mean in this context.

However from a sighted perspective, it could be a little strange to have different icons used for the same "Approved" status.

> Table header: Is there an option to enable the header sections to organize the table content (Like the filter in Excel?)

While there's no built-in functionality in the Table component yet, it is possible for your team to add the functionality on top of the component. Let us know if you end up going this route, and if you'd like to collaborate on the design and what icons to use!

> Filter: Does it affects the accessibility part the fact of not having a 'filter' button? Is there a pattern for this?

I don't think there's a hard accessibility rule about not having a button on filter form fields. However adding a "filter" button could generally increase clarity about when something on the page changes and gives the customer more control over when the filtering happens.

We've generally seen the "Filter" button used across most tables in Twilio products—my guess is because making an API call for each form field change would be costly and the loading experience on each change might not be a great customer experience. I'd definitely check with your product team to test that out for this page.

> Call to action on the form of a menu displaying three actions: We are migrating this to the Paste components, but I'm not sure if this action menu exists in Paste, or if there is an alternative to have 3 action items in one row of a table.

Let's chat more about this tomorrow! I don't see a menu in the prototype you shared. If you mean the "More" icon in the table under "Supporting Documents", that's definitely available in Paste. If there's an action item in there that's really critical to the customer's goal, I might suggest  pulling it out, but since there are already a lot of actions on this screen, it'd be great to hear how you're thinking of approaching these actions in the context of the entire page.

**Comment by SiTaggart** [comment]
Link to the recording of office hours https://www.loom.com/share/7fccefc339fc4b1caf9710961c648fdb

Passcode can be found in the [Slack thread](https://twilio.slack.com/archives/CBEQHHDLP/p16210194101589000
